crt_portal/tms/admin.py

```diff
@@ -1,10 +1,39 @@
-from cts_forms.admin import ReadOnlyModelAdmin
+import logging
+from cts_forms.admin import ReadOnlyModelAdmin, csv, Echo, iter_queryset, format_export_message
 from django.contrib import admin
 from django.utils.html import format_html
 from django.urls import reverse
+from django.http import StreamingHttpResponse
 
 from .models import TMSEmail
 
+TMS_FIELDS = ['id', 'tms_id', 'subject', 'body', 'recipient', 'status', 'error_message', 'report_id', 'created_at', 'completed_at', 'purpose']
+
+logger = logging.getLogger(__name__)
+
+
+def _serialize_tms(data):
+    """Preserve headers while rendering TMS_FIELDS for inbound actions"""
+    if isinstance(data, TMSEmail):
+        return [getattr(data, field) for field in TMS_FIELDS]
+    return data
```

Comment:
This feels like it could turn into a bug, maybe? If something is not an instance of the TMSEmail model, I feel like we should return an exception, right? We're trying to serialize a very specific thing, and I think if the data coming in isn't that thing, then we shouldn't serialize it? Happy to chat more.

Reply:
I am copying the pattern over in cts_forms/admin.  My guess is they coded it that way so if you added new fields to the model, it would not break if python had trouble reading those fields, like if it was something complicated.  (see how they are doing it in reports)

This _serialize function is only for tms models though.  Do you think we should extend it to be more general.  The pattern so far is to have a serialize function and export function that is specific for each model.  I personally think that is the right way to go for clarity (like report is different than actions).  

What do you think?

Reply:
Sounds good! This answers my question. Thanks @billyhunt